### PR TITLE
add few vim resources

### PR DESCRIPTION
**Pull request by techgaun**
## What does this PR do?
Add Resource(s)

### Description 
Added two books that are free and open-source.

### Why is this valuable (or not)

Both of these online references are great introductions to vim.

### How do we know it's really free?

Based on their github pages.

### For book lists, is it a book?

both of them are not the standard traditional books but similar to some of the other stuff here to count as a book imo.

### Checklist:
- [x] Not a duplicate
- [ ] Included author(s) if appropriate
- [x] Lists are in alphabetical order
- [x] Needed indications added (PDF, access notes, under construction)

### CODE REVIEW

File: free-programming-books.md [hunk 1]
```diff
@@ -2740,9 +2740,11 @@ Kerridge (PDF) (email address *requested*, not required)
 ### Vim
 
 * [A Byte of Vim](http://www.swaroopch.com/notes/vim/)
+* [Learn Vim (the Smart Way)](https://github.com/iggredible/Learn-Vim) (HTML)
```
Comment (eshellman) [inline free-programming-books.md:2743]: says it's "in progress". We have a notation for that!

**Comment by eshellman** [comment]
... and the linter disagrees about the list order.

File: free-programming-books.md (hunk 1 above)
Comment (techgaun) [reply free-programming-books.md:2743]: @eshellman the book is very well in progress and its evolving but even then, the content there is right now is a lot more helpful than many of the other resources. But, if you think this still can not make it to the list, I am happy to remove it. let me know what you think.

Comment (eshellman) [reply free-programming-books.md:2743]: add it with the "in progress" notation. please read /CONTRIBUTING.md !!!

Comment (techgaun) [reply free-programming-books.md]: oops sorry updated now accordingly

**Comment by bigbirdone** [comment]
zn15in

在 2020-10-12 10:03:01，"Eric Hellman" <notifications@github.com> 写道：


Merged #3882 into master.

**Pull request merged**